Assets/Scripts/Game/InputManager.cs

```diff
@@ -518,6 +766,90 @@ public void ResetDefaults()
 
             SetBinding(KeyCode.F9, Actions.QuickSave);
             SetBinding(KeyCode.F12, Actions.QuickLoad);
+
+            SetAxisBinding("Axis1", AxisActions.MovementHorizontal);
+            SetAxisBinding("Axis2", AxisActions.MovementVertical);
+            SetAxisBinding("Axis4", AxisActions.CameraHorizontal);
+            SetAxisBinding("Axis5", AxisActions.CameraVertical);
+
+            SetAxisBinding("Axis9", AxisActions.LeftTrigger);
+            SetAxisBinding("Axis10", AxisActions.RightTrigger);
+            SetAxisBinding("Axis6", AxisActions.DPadHorizontal);
+            SetAxisBinding("Axis7", AxisActions.DPadVertical);
+
+            controllerUIDict[0] = KeyCode.JoystickButton0;
+            controllerUIDict[1] = KeyCode.JoystickButton1;
+            UpdateAxisBindingCache();
+        }
+
+        public bool GetMouseButtonDown(int button)
+        {
+            if (usingControllerCursor)
+                return GetKeyDown(controllerUIDict[button]);
+
+            return Input.GetMouseButtonDown(button);
+        }
+
+        public bool GetMouseButton(int button)
+        {
+            if (usingControllerCursor)
+                return GetKey(controllerUIDict[button]);
+
+            return Input.GetMouseButton(button);
+        }
+
+        public bool GetKey(KeyCode key)
+        {
+            KeyCode conv = ConvertJoystickButtonKeyCode(key);
+            var k = (((int)conv) < 5000 && Input.GetKey(conv)) || GetAxisKey((int)conv);
+            if (k)
+                LastKeyDown = conv;
+            return k;
+        }
+
+        public bool GetKeyDown(KeyCode key)
+        {
+            KeyCode conv = ConvertJoystickButtonKeyCode(key);
+            var kd = (((int)conv) < 5000 && Input.GetKeyDown(conv)) || GetAxisKeyDown((int)conv);
+            if (kd)
+                LastKeyDown = conv;
+            return kd;
+        }
+
+        public bool GetKeyUp(KeyCode key)
+        {
+            KeyCode conv = ConvertJoystickButtonKeyCode(key);
+            return (((int)conv) < 5000 && Input.GetKeyUp(conv)) || GetAxisKeyUp((int)conv);
+        }
+
+        public bool AnyKeyDown
+        {
+            get
+            {
+                foreach (KeyCode k in KeyCodeList)
+                    if (GetKeyDown(k)) return true;
+                return false;
+            }
+        }
+
+        public String GetKeyString(KeyCode key)
+        {
+            if (axisKeyCodeStrings.ContainsKey((int)key))
+                return axisKeyCodeStrings[(int)key];
+            else
+                return key.ToString();
+        }
+
+        public KeyCode ParseKeyCodeString(String s)
+        {
+            try
+            {
+                return (KeyCode)Enum.Parse(typeof(KeyCode), s);
+            }
+            catch
+            {
+                return (KeyCode)axisKeyCodeStrings.FirstOrDefault(x => x.Value == s).Key;
+            }
```

Comment:
I would use `Enum.IsDefined(typeof(KeyCode), s)` to prevent exceptions instead of catching them.

Reply:
Thanks, I'll adjust when I get home.